fdbsyncd/fdbsync.h

```diff
@@ -10,7 +10,16 @@
 #include "warmRestartAssist.h"
 
 // The timeout value (in seconds) for fdbsyncd reconcilation logic
-#define DEFAULT_FDBSYNC_WARMSTART_TIMER 30
+#define DEFAULT_FDBSYNC_WARMSTART_TIMER 600

```

Comment:
What is the purpose of increasing this timer. @zhenggen-xu can you also review?

Reply:
The timer seems to be big, so we need 10 minutes for FDB to sync? how many entries and how do we get to that number?

Reply:
This is the max timer to wait for orchagent reconciliation.  This will not be hit in normal circumstances. This is just to avoid indefinite wait on orchagent to reconcile and hence a large value. The code will not wait for this timer once the orchagent had reconciled.

Reply:
Sounds like there were many timers, could you make a inline comments in the code to describe the relationships between different timers?

Reply:
Reconciliation and restoration timers are not actually related, except that the reconciliation starts after restoration ends. And the warm-start timer is like fallback timeout to avoid indefinite wait for reconciliation. Will try to include some more description there

Reply:
This is the first time warm-reboot is implemented for fdbsyncd. So DEFAULT_FDBSYNC_WARMSTART_TIMER=30 was not really a value that was used.  Please note that DEFAULT_FDBSYNC_WARMSTART_TIMER=600 is a fallback timer to proceed with system operation if the dependency wait does not resolve for long time. This will make sure that even if warm-reboot fails, system will not wait indefinitely and will recover. FDBSYNC_RECON_WAIT_TIME is 120 is based in the default value of reconciliation time for fpmsyncd.

Reply:
To me, we are mixing things here.  FDBSYNC_RECON_WAIT_TIME in your code is the timer to wait for orchagent to be RECONSILED,  that does not depending on other application timers.  I don't see why it should be 120 based on fpmsyncd. Do we have a test log that showing the events in sequence?

Reply:
FDBSYNC_RECON_WAIT_TIME is not the timer to wait for orchagent to reconcile. FDBSYNC_RECON_WAIT_TIME is the time for control plane ( BGP) to re-establish and reconcile its data ( routes) which are then programmed to fpmsyncd/fdbsyncd and hence the time given is same as fpmsyncd.

Reply:
See your fdbsyncd.cpp in your PR:
line 110:                         reconCheckTimer.setInterval(timespec{FDBSYNC_RECON_WAIT_TIME, 0});
line 124: if (sync.isReadyToReconcile())

and https://github.com/Azure/sonic-swss/blob/5384ace1440cb97cd4829e416e26b06f0361c9f8/orchagent/orchdaemon.cpp#L671

Reply:
Did not understand the comment completely. Can you please elaborate a little ?  FDBSYNC_RECON_WAIT_TIME  is the time we are waiting for the control plane ( in this case BGP ) to converge after restart and after this we also need to check if orchagent has reconciled after warm-reboot. Hence when this timer FDBSYNC_RECON_WAIT_TIME  expires, we check if orchagent is reconciled and hereon we keep checking for orchagent to be reconciled so that we can mark fdbsyncd as reconciled and it can start updating the APP-DB.

Reply:
The code or even comments in the code does not specify the relationship between this FDBSYNC_RECON_WAIT_TIME timer and BGP timer.  If this process has dependency on BGP reconciliation, we should make it explicit, e,g, fpmsyncd could write a flag to somewhere (e,g stateDB) and this process can pick it up.  Having a hardcoded timer here to assume the BGP has the same timer (BGP timer, which is configurable) does not seem to be a good idea.